core/templates/services/user.service.ts

```diff
@@ -56,21 +60,33 @@ export class UserService {
     return this.userInfo;
   }
 
-  async getProfileImageDataUrlAsync(): Promise<string> {
-    let defaultUrl = (
-      this.urlInterpolationService.getStaticImageUrl(
-        AppConstants.DEFAULT_PROFILE_IMAGE_PATH));
-    return this.getUserInfoAsync().then(
-      async(userInfo) => {
-        if (userInfo.isLoggedIn()) {
-          return this.userBackendApiService.getProfileImageDataUrlAsync(
-            defaultUrl);
-        } else {
-          return new Promise((resolve, reject) => {
-            resolve(defaultUrl);
-          });
-        }
-      });
+  getProfileImageDataUrlAsync(
+    username: string, isWebp: boolean = false): string {
+    let defaultUrlWebp = this.urlInterpolationService.getStaticImageUrl(
+      AppConstants.DEFAULT_PROFILE_IMAGE_WEBP_PATH);
+    let defaultUrlPng = this.urlInterpolationService.getStaticImageUrl(
+      AppConstants.DEFAULT_PROFILE_IMAGE_PNG_PATH);
+    let localStoredImage = this.imageLocalStorageService.getRawImageData(
+      username + '_profile_picture.png');
+    if (localStoredImage !== null) {
+      return localStoredImage;
+    }
+    if (AppConstants.EMULATOR_MODE) {
+      if (isWebp) {
+        return defaultUrlWebp;
+      }
+      return defaultUrlPng;
+    }
+    else {
+      if (isWebp) {
+        return this.urlInterpolationService.interpolateUrl(
+          this.assetsBackendApiService.profileImageWebpUrlTemplate,
+          {username: username})
+      }
+      return this.urlInterpolationService.interpolateUrl(
+        this.assetsBackendApiService.profileImagePngUrlTemplate,
+        {username: username})
+    }
```

Comment:
I was also thinking about this part but for the part where we have stored the profile picture in the session storage we are only returning that picture so we cannot always return tuple.
Thinking of this what we can do is to also return the tuple for the images stored in localStorage. So it will be something like return (localStoredImage, localStoredImage). Does this approach sounds good to you?
Thanks

Reply:
I think it is fine to return a tuple with the same element(PNG image) for the specific case.

Reply:
Done